pkg/k8sutil/k8sutil.go

```diff
@@ -83,3 +87,32 @@ func ResourceExists(dc discovery.DiscoveryInterface, apiGroupVersion, kind strin
 	}
 	return false, nil
 }
+
+// GetPod returns a Pod object that corresponds to the pod in which the code
+// is currently running.
+// It expects the environment variable POD_NAME to be set by the downwards API.
+func GetPod(ctx context.Context, client crclient.Client, ns string) (*corev1.Pod, error) {
+	podName := os.Getenv(PodNameEnvVar)
+	if podName == "" {
+		return nil, fmt.Errorf("required env %s not set, please configure downward API", PodNameEnvVar)
+	}
+
+	log.V(1).Info("Found podname", "Pod.Name", podName)
+
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
```

Comment:
This suggestion isn't strictly necessary but using `corev1.SchemaGroupVersion.Version` is a better way of setting this value IMO. WDYT?

Reply:
Is it necessary to set the TypeMeta for a concrete type when using the controller-runtime client? I thought this was only necessary with `unstructured.Unstructured`.

Reply:
No idea why this was done this way. I can remove it, but it comes originally and is used still in the `leader.go` `myOwnerRef` function, so not sure if they need to set this maybe. I would leave it, as it doesn't hurt. https://github.com/operator-framework/operator-sdk/pull/1037/files#diff-a0eaeead6981b3b2f753c1ecbba711ccL145